outlines/models/openai.py

```diff
@@ -17,6 +17,20 @@
     from openai import AsyncOpenAI
 
 
+@dataclass(frozen=True)
+class OpenAIUsage:
+    prompt_tokens: Optional[int] = 0
+    completion_tokens: Optional[int] = 0
+    total_tokens: Optional[int] = 0
```

Comment:
Since pricings are different it's probably unnecessary

Reply:
That's probably right. I think prices used to be different for the "completion" endpoints, not chat completion. If you want to just track total_tokens, then it simplifies quite a bit.  No class, and the call to async should just return the int.

Reply:
On a closer look, prompt and response tokens have indeed different costs: https://openai.com/pricing. 
Hence, if at all, I would return to my previous commit -1, where cost were kept separately.

Reply:
In which case we don't need the total number of tokens

Reply:
If only they had an API that returns the price per token for each model...

Reply:
Yep, no luck there. What would be your preferred solution to get this or something similar merged? 
Since we have to count different kinds of tokens, I would prefer the method, to avoid having the code for the token count update in two places.
Whether we need a class or two instance attributes, that's up to you.
At some point, I would like publish a repo where I benchmark guided vs. unguided generation in terms of cost, and would be nicer if I can add a dependency to the released version of outlines.